### ISSUE TITLE
are all the field ids necessary?

### ISSUE THREAD

**Issue by vinniefalco** [cppalliance/http#29]
We might remove most of the field ids and their corresponding strings in the lookup table, and people would likely not care as long as we keep the most commonly used ones. Another possibility is to make the lookup a function pointer or abstract class, so that it can be substituted by the user.

**Comment by sgerbino** [comment]
Done in a100477, which curated the field enum down to modern HTTP fields, removing the rarely used ids.

**Comment by vinniefalco** [comment]
did we really gain anything by removing these?

**Comment by ashtum** [comment]
> did we really gain anything by removing these?

A lot of the ids were related to the SMTP protocol and unrelated to an HTTP library. A good portion of parsing time is spent mapping each field name to its id (using the perfect hash map). The list is reduced to 130 items instead of 357 without losing anything meaningful.
The perfect hash map now uses a much smaller table:
```cpp
- enum { N = 5155 };
- unsigned char map_[ N ][ 2 ] = {};
+ enum { N = 1367 };
+ unsigned char map_[ N ] = {};
```
This also reduces visual clutter when someone is looking for a field in the list (through IDE code completion) and adds items that were not in the list but were included in the standard.

**Comment by vinniefalco** [comment]
The hash map was tuned for a specific data set and now the data set changed, did we recalculate the perfect hash? What's the fill/collision rate now?

**Comment by ashtum** [comment]
> The hash map was tuned for a specific data set and now the data set changed, did we recalculate the perfect hash? What's the fill/collision rate now?

Yes, that's how `1367` is calculated. By the way, this change was also applied to Beast in Boost 1.90 release.
